### DISCUSSION TITLE
Running executables with steel

### DISCUSSION THREAD

**Discussion by nxy7** [mattwparas/steel#202] [Q&A] [answered]
Hello, I've never used lisp languages so I don't really know how to write them very well. I usually learn by doing and I think that neat usecase for me would be swapping shell scripts with steel ones. I think that maybe macro system could provide good ergonomics here that regular programming languages I use are lacking (go, rust, etc).

Is there any example of how to run some program, capture it's output etc. etc.? I've looked through examples and didn't find anything but maybe I'm just blind :P

**Comment by mattwparas** [comment, marked as answer]
So you'll find the process module [here](https://github.com/mattwparas/steel/blob/master/crates/steel-core/src/primitives/process.rs)

Here is an example of running a process:

```scheme
(define (build-release)
  (~> (command "cargo" '("build" "--release")) (spawn-process) (Ok->value) (wait)))
```

To capture std out you could try:

```scheme
(define (build-release)
  (~> (command "cargo" '("build" "--release")) (spawn-process) (Ok->value) (wait->stdout)))
```

It certainly could use some documentation 😅
